src/liballoc/vec.rs

```diff
@@ -813,14 +813,19 @@ impl<T> Vec<T> {
             for i in 0..len {
                 if !f(&v[i]) {
                     del += 1;
+                    unsafe {
+                        ptr::read(&v[i]);
```

Comment:
Shouldn't this be a call to drop_in_place?

Reply:
Ah, yes it should.  I didn't know that function existed, so thank you!